### PR TITLE
Chore/Release process setup

### PR DESCRIPTION
**Pull request by GregBal**
On this PR, the publish plugins and the documentation for the release process has been added,
along with a github action responsible for uploading kotlinDocs on github pages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,3 +148,24 @@ class OutboxMonitor(
   }
 }
 ```
+
+## Publishing
+
+* Bump version in `gradle.properties` of `kfactory` module.
```
Comment (ippokratoys) [inline README.md:154]: kfactory
image

Comment (GregBal) [reply README.md:154]: ahahahaha

File: core/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,20 @@
+GROUP=io.github.bluegroundltd
+POM_ARTIFACT_ID=transactional-outbox-core
+VERSION_NAME=0.0.1
+
+POM_NAME=Transactional Outbox Core
+POM_DESCRIPTION=Easily implement the transactional outbox pattern in your java application
```
Comment (chris-asl) [inline core/gradle.properties:6]: extreme-nit: in your JVM application 😉

File: README.md [hunk 3]
```diff
@@ -12,14 +12,15 @@ application.
   - [Creating a new outbox handler](#creating-a-new-outbox-handler)
   - [Creating a new outbox entry](#creating-a-new-outbox-entry)
   - [Monitoring the outbox entries](#monitoring-the-outbox-entries)
+- [Publishing](#publishing)
```
Comment (chris-asl) [inline README.md:15]: I think that sth is off here 👀 (the `+` and indentation)

File: README.md [hunk 4]
```diff
@@ -147,3 +148,24 @@ class OutboxMonitor(
   }
 }
 ```
+
+## Publishing
+
+* Bump version in `gradle.properties` of `kfactory` module.
+* Execute the following to upload artifact:
+```shell
+$ ./gradlew :kfactory:publish \
```
Comment (chris-asl) [inline README.md:157]: Commenting to make sure we don't miss the `kfactory` from here, too

**Comment by chris-asl** [comment]
Hmm, have we configured required approvals for PRs? 
It seems we can merge this, despite no approvals.

**Comment by GregBal** [comment]
> Hmm, have we configured required approvals for PRs? It seems we can merge this, despite no approvals.

It is targeting another branch that's why, we 've got the checks when merging on main

File: README.md (hunk 3 above)
Comment (GregBal) [reply README.md:18]: Hmmm I can't see actually what's off

Comment (chris-asl) [reply README.md:18]: You're right! I thought we had a single level of indentation here

**Review by chris-asl — APPROVED** [review]
LGTM! 🚀

**Pull request merged**